docs/concepts/controllers/statefulsets.md

```diff
@@ -0,0 +1,169 @@
+---
+assignees:
+- bprashanth
+- enisoc
+- erictune
+- foxish
+- janetkuo
+- kow3ns
+- smarterclayton
+---
+
+{% capture overview %}
+**Stateful Sets are a beta feature in 1.5. This feature replaces the 
+Pet Sets feature from 1.4. Users of Pet Sets are referred to the 1.5 
+[Upgrade Guide](/docs/tasks/stateful-set/upgrade-from-petsets-to-stateful-sets/)
+for further information on how to upgrade existing Pet Sets to Stateful Sets.**
+
+A Stateful Set is a Controller that provides a unique identity to its Pods. It provides
+guarantees about the ordering of deployment and scaling.
+{% endcapture %}
+
+{% capture body %}
+
+### When to Use a Stateful Set
+Stateful Sets are valuable for applications that require one or more of the 
+following.
+
+* Stable, unique network identifiers.
+* Stable, persistent storage.
+* Ordered, graceful deployment and scaling.
+* Ordered, graceful deletion and termination.
+
+In the above, by stable, we mean persistent across Pod (re) schedulings.
+As it is generally easier to manage, if an application doesn't require any of 
+the above guarantees, and if it is feasible to do so, it should be deployed as 
+a set of stateless replicas.
+
+### Limitations
+* Stateful Set is a beta resource, not available in any Kubernetes release prior to 1.5.
+* As with all alpha/beta resources, it can be disabled through the `--runtime-config` option passed to the apiserver.
+* The storage for a given Pod must either be provisioned by a [Persistent Volume Provisioner](http://releases.k8s.io/{{page.githubbranch}}/examples/experimental/persistent-volume-provisioning/README.md) based on the requested `storage class`, or pre-provisioned by an admin.
+* Deleting and/or scaling a Stateful Set down will *not* delete the volumes associated with the Stateful Set. This is done to ensure safety first, your data is more valuable than an auto purge of all related Stateful Set resources. 
+* Stateful Sets currently require a [Headless Service](/docs/user-guide/services/#headless-services) to be responsible for the network identity of the Pods. The user is responsible for this Service.
+* Updating an existing Stateful Set is currently a manual process, meaning you either need to deploy a new Stateful Set with the new image version, or orphan Pods one by one, update their image, and join them back to the cluster.
```

Comment:
Maybe just say: "Updating an existing Stateful Set is currently a manual process: the pod template cannot be updated.".  Leave the explanation of how you might manually update for later.